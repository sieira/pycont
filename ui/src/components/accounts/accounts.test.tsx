import React from 'react'
import AccountList from './accounts'

import { createWithProvider } from '../../test-utils/render-utils.test'

describe('Accounts tests', () => {
  beforeEach(() => {
    fetch.resetMocks()
  })

  it('Should show accounts if fetched', () => {
    const accounts = createWithProvider(<AccountList />, {
      accounts: {
        accountList: [{ name: 'Account1' }, { name: 'Account2' }],
        fetched: true,
      },
    })
    expect(accounts).toMatchSnapshot()
  })

  it('Should show loader if not fetched', () => {
    fetch.mockResponseOnce(JSON.stringify([]))
    const accounts = createWithProvider(<AccountList />, {
      accounts: {
        accountList: undefined,
        fetched: false,
      },
    })
    expect(accounts).toMatchSnapshot()
  })
})
