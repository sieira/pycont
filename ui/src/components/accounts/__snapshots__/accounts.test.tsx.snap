// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts tests Should show accounts if fetched 1`] = `
Array [
  <h2>
    Accounts
  </h2>,
  <div
    aria-busy={false}
    aria-live="polite"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        <span>
          Account1
        </span>
        <span />
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        <span>
          Account2
        </span>
        <span />
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`Accounts tests Should show loader if not fetched 1`] = `
Array [
  <h2>
    Accounts
  </h2>,
  <div
    aria-busy={true}
    aria-live="polite"
  >
    <div
      className="spinner-grow text-info"
    />
  </div>,
]
`;
